### PR TITLE
made repo pip installable

### PR DESCRIPTION
**Pull request by huaijiangzhu**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

## How I Tested

[//]: # "Explain how you tested your changes"

## Do not merge before

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

- [ ] link to dependent pull request

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [ ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [ ] All new functions/classes are documented and existing documentation is updated according to changes.
- [ ] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: demos/demo_display_solo12_meshcat.py
Comment (jviereck) [inline demos/demo_display_solo12_meshcat.py]: nit: missing new line at end of the file.

File: demos/demo_display_solo8_meshcat.py
Comment (jviereck) [inline demos/demo_display_solo8_meshcat.py]: nit: missing new line at end of the file.

File: readme.md
Comment (jviereck) [inline readme.md:75]: Seems like you have been to the future already ;) Usually you use only up to the year of the last edit.

File: src/robot_properties_solo/config.py
Comment (jviereck) [inline src/robot_properties_solo/config.py]: how about calling this `dgm_yaml_path` - same below?

File: src/robot_properties_solo/utils.py
Comment (jviereck) [inline src/robot_properties_solo/utils.py:17]: Do we support python2 still? If not, I would remove the try-except block and just have the python3 code.

**Review by jviereck — APPROVED** [review]

**Review by jviereck — APPROVED** [review]

**Comment by jviereck** [comment]
Thanks for your contribution! Approving with a few small comments.

I don't see a PR for `bullet_utils` yet. Are you planning to create a pull request for `pinocchio_bullet` as well and then rename the repo?

File: src/robot_properties_solo/config.py
Comment (huaijiangzhu) [reply src/robot_properties_solo/config.py]: I agree. I didn't do it because I wasn't sure if this would break up people's code.

**Comment by huaijiangzhu** [comment]
> Thanks for your contribution! Approving with a few small comments.
> 
> I don't see a PR for `bullet_utils` yet. Are you planning to create a pull request for `pinocchio_bullet` as well and then rename the repo?

Max and I agreed that we will simply create a new repo for bullet_utils to spare the hustle. What do you think?

File: readme.md
Comment (huaijiangzhu) [reply readme.md:75]: 2020 is canceled ;-)

File: src/robot_properties_solo/config.py
Comment (righetti) [inline src/robot_properties_solo/config.py:131]: with the change, this throws an error on my computer
`ValueError: could not broadcast input array from shape (15) into shape (15,1)`
I suggest changing into
```suggestion
    q0 = initial_configuration
```
this solved my issue

File: src/robot_properties_solo/config.py
Comment (righetti) [inline src/robot_properties_solo/config.py:231]: same pb here as above suggest to change as
```suggestion
    q0 = initial_configuration
```

File: src/robot_properties_solo/config.py
Comment (huaijiangzhu) [reply src/robot_properties_solo/config.py:131]: This seems to be caused by the change of array convention in pinocchio. Is your pinocchio up-to-date? The problem of the suggested change is that it makes q0 a list instead of a numpy array.

Comment (righetti) [reply src/robot_properties_solo/config.py:131]: I have pinocchio 2.5.2 - is it too old? but good point, my fix is potentially an issue too...

Comment (huaijiangzhu) [reply src/robot_properties_solo/config.py:131]: That's strange. I have the same version but I don't get this error. When I call pinocchio.utils.zero(n), I get a numpy array of shape (n,) instead of (n, 1), is this the same case with you? Previously, the default shape was (n, 1) but LAAS changed it at some point.
At the end of the day, I'd prefer a solution that works with both shapes to not break apart some legacy code written in old pinocchio convention, but maybe that's not necessary.

Comment (righetti) [reply src/robot_properties_solo/config.py:131]: ok - when I start an isolated python session and call pinocchio.utils.zero(n) I get shape(n,)
however in the robot_properties function, I get shape(n,1).
I checked and versions of pinocchio and numpy are the same in both cases but the same commands have different outputs. I do not know why.
Do you see the same thing?

File: src/robot_properties_solo/config.py
Comment (jviereck) [reply src/robot_properties_solo/config.py]: Oh, I see, it was there before. Then maybe let's keep it for now.

**Comment by jviereck** [comment]
> Max and I agreed that we will simply create a new repo for bullet_utils to spare the hustle. What do you think?

That's fine with me. We should have the new bullet_utils in place before merging this PR.

File: src/robot_properties_solo/resources/xacro/leg.xacro
Comment (wxmerkt) [inline src/robot_properties_solo/resources/xacro/leg.xacro]: Doesn't omitting the package name break the ability to visualise in RViz and other viewers? For PyBullet, you can add additional search paths so it works with standard URDFs out of the box

Comment (MaximilienNaveau) [reply src/robot_properties_solo/resources/xacro/leg.xacro]: Yes this most likely break it... I thought I made sure the urdf was able to load in both RVIZ+GepttoViewer+PyBullet
I would like to preserve this feature please.

**Review by righetti — COMMENTED** [review]
I found the issue - in the test code of kino-dyn-opt, we use ```eigenpy.switchToNumpyMatrix()``` and ```pinocchio.switchToNumpyMatrix()```. If I comment this out all the tests pass without problem and the error is not raised anymore. However I do not know the consequences of this for the rest of the code.
Anyone has any thoughts on this?

**Comment by jviereck** [comment]
> I found the issue - in the test code of kino-dyn-opt, we use eigenpy.switchToNumpyMatrix() and pinocchio.switchToNumpyMatrix(). If I comment this out all the tests pass without problem and the error is not raised anymore. However I do not know the consequences of this for the rest of the code.
Anyone has any thoughts on this?

Our code is intended to work with `eigenpy.switchToNumpyArrat()` by default by now (which is also the default by pinocchio and what we agreed on a while back). If you remove the line and thing still work, then it's all good ;)

Some background what goes on under the hood: Eigenpy allows to return `np.matrix` or `np.array` for Eigen matrixes. It turns out that returning `np.array` is 1) faster and 2) the future as the use of `np.matrix` is deprecated. It used to be the default that eigenpy/pinocchio returned `np.matrix`es. This is no longer the default and it returns `np.array`s by now.

**Comment by MaximilienNaveau** [comment]
> switchToNumpyMatrix

You are right this is the bug!
This was used for historical reasons but must be deleted (see previous message of Julian)
The last versionS of pinocchio switched to numpy.array hence should be compatible with the config files.

File: src/robot_properties_solo/resources/xacro/leg.xacro
Comment (huaijiangzhu) [reply src/robot_properties_solo/resources/xacro/leg.xacro]: I see! Thanks for pointing it out.

**Pull request merged**
